src/Conversion/KrnlToLLVM/KrnlPrint.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------ KrnlPrint.cpp - Lower KrnlPrintOp -----------------------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// This file lowers the KrnlPrintOp operator.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+
+#include "src/Conversion/KrnlToLLVM/KrnlPrint.hpp"
+#include "src/Conversion/KrnlToLLVM/KrnlToLLVM.hpp"
+#include "src/Conversion/KrnlToLLVM/KrnlToLLVMHelper.hpp"
+#include "src/Dialect/Krnl/KrnlHelper.hpp"
+#include "src/Dialect/Krnl/KrnlOps.hpp"
+
+#include "llvm/Support/Debug.h"
+
+#define DEBUG_TYPE "krnl_to_llvm"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+static int64_t count = 0;
```

Comment:
Is a good practice to use a mutable static local variable in a pass? It is OK since this is not critical code.

Reply:
It is not for critical code because it might force serialization and global side effects. In this particular case I need to give a unique name to the format specifier used by the printf call and so I think this use is justifiable. Happy to change this if there is a suggestion to help avoid the static global var.

Reply:
I ended up finding a way to remove the global static variable.